JetNews/app/src/main/java/com/example/jetnews/ui/interests/InterestsScreen.kt

```diff
@@ -510,6 +507,46 @@ private fun InterestsTabRow(
     }
 }
 
+/**
+ * Return state for interest sections grouped given the number of columns that fill the screen.
+ * Sections are distributed to balance the number of items on each column.
+ */
+@Composable
+private fun sectionsGroupedInColumnsAsState(
+    sections: List<InterestSection>,
+    columns: Int,
+    defaultDispatcher: CoroutineDispatcher = Dispatchers.Default
+): State<Array<MutableList<InterestSection>>> = produceState(
+    initialValue = arrayOf(),
+    keys = arrayOf(sections, columns)
+) {
+    withContext(defaultDispatcher) { // Move the execution of this logic to the background thread
+        val sectionsInColumns = Array<MutableList<InterestSection>>(columns) { mutableListOf() }
+        // For each section, find the column with less elements and add it there
+        sections.forEach { section ->
+            val shortestColumnIndex = sectionsInColumns.getIndexShortestElement()
```

Comment:
I think we can avoid having to search through all of the existing `sectionsInColumns` if we keep a running tally of the number of elements in each column. I think that should make it a bit more efficient as well (on the order of `O(sections.size * columns)` instead of `O(sections.size^2 * columns)`)

If we do that, we probably don't need to put this in the background thread either? It shouldn't be substantially slower that the older grouping method.

Reply:
Done thanks!